app/src/pages/inside/stepPage/modals/editDefectModals/makeDecisionModal/makeDecisionModal.jsx

```diff
@@ -43,77 +43,110 @@ import {
 } from 'controllers/plugins';
 import { getIssueTitle } from 'pages/inside/common/utils';
 import { LINK_ISSUE, POST_ISSUE, UNLINK_ISSUE } from 'common/constants/actionTypes';
+import { ExecutionInfo } from 'pages/inside/logsPage/defectEditor/executionInfo';
+import { historyItemsSelector } from 'controllers/log';
 import { ActionButtonsBar } from './actionButtonsBar';
 import { messages } from './../messages';
-import { MAKE_DECISION_MODAL, CONFIGURATION, OPTIONS } from '../constants';
-import styles from './makeDecisionModal.scss';
+import {
+  CONFIGURATION,
+  COPY_FROM_HISTORY_LINE,
+  MACHINE_LEARNING_SUGGESTIONS,
+  MAKE_DECISION_MODAL,
+  OPTIONS,
+  SELECT_DEFECT_MANUALLY,
+} from '../constants';
 import { OptionsStepForm } from './optionsStepForm';
+import styles from './makeDecisionModal.scss';
 
 const cx = classNames.bind(styles);
 
+const TABS_ID = {
+  [MACHINE_LEARNING_SUGGESTIONS]: 0,
+  [COPY_FROM_HISTORY_LINE]: 1,
+  [SELECT_DEFECT_MANUALLY]: 2,
+};
+
 const MakeDecision = ({ data }) => {
   const { formatMessage } = useIntl();
   const dispatch = useDispatch();
   const activeProject = useSelector(activeProjectSelector);
   const btsIntegrations = useSelector(availableBtsIntegrationsSelector);
   const isBtsPluginsExist = useSelector(isBtsPluginsExistSelector);
   const enabledBtsPlugins = useSelector(enabledBtsPluginsSelector);
+  const historyItems = useSelector(historyItemsSelector);
   const isPostIssueUnavailable = !isPostIssueActionAvailable(btsIntegrations);
   const isBulkOperation = data.items && data.items.length > 1;
   const itemData = isBulkOperation ? data.items : data.items[0];
-  const [state, setState] = useState({
+  const [selectDefectTypeState, setSelectDefectTypeState] = useState({
     issue: isBulkOperation ? {} : itemData.issue,
   });
-  const [modalHasChanges, setModalHasChanges] = useState(false);
+  const [copyFromHistoryState, setCopyFromHistoryState] = useState({});
```

Comment:
It looks like these actions provide similar information about the item and its defect. Do we really need to keep them in different states? What if we add a new section to the accordion - this requires a new useState call?
Can we just store them in one with the source type specified in the additional field, for example `source: COPY_FROM_HISTORY_LINE`.
It will also help remove unnecessary checks and reduce processing logic.